src/Icicle/Source/ToCore/ToCore.hs

```diff
@@ -637,19 +638,21 @@ convertReduce xx
 
 
  | Case (Annot { annAnnot = ann, annResult = retty }) scrut patalts <- xx
- = do   scrut' <- convertReduce scrut
-        let (pats,alts) = unzip patalts
-        alts'  <- mapM convertReduce alts
+ = do   let (pats,alts) = unzip patalts
+
+        scrut' <- convertReduce scrut
+        pats'  <- convertCaseFreshenPats pats
+        alts'  <- mapM convertExp alts
+        alts'' <- hoist runFreshIdentity $ lift $ mapM (ANormal.anormal ()) alts'
```

Comment:
I wanted to make sure nested case statements get generated to `conv = let .. in .. let .. in ....`. Perhaps that is the case even without a-normalisation... This line might be premature, yeah.